GH-15780 set weak learner parameter api documentation (#15920)
h2oai/h2o-3: h2o-docs/src/product/data-science/adaboost.rst

      - ``GLM``: Trains a binary classifier with ``max_iterations=50``.

      - ``DEEP_LEARNING``: Trains a binary classifier with ``(epochs=10, hidden=[2])``.

> -  **weak_learner_params**:  With ``weak_learner``, you can also specify a dict/list of customized parameters for that algorithm. For example if we use a ``GBM``, we can specify ``{'ntrees': 1, 'max_depth': 10}`` in Python and ``list(ntrees = 1, max_depth = 10)`` in R. AdaBoost does not apply defaults from the ``weak_learner`` at all. The defaults of the algorithm itself will be used instead.

Reviewer: ```suggestion
-  **weak_learner_params**:  You can specify a dict/list of customized parameters for your specified ``weak_learner`` algorithm. For example, if you use a ``GBM``, you can specify ``{'ntrees': 1, 'max_depth': 10}`` in Python or ``list(ntrees = 1, max_depth = 10)`` in R. AdaBoost does not apply defaults from the ``weak_learner`` at all. The defaults of the algorithm itself will be used instead.
```
